src/main.rs

```diff
@@ -542,18 +541,18 @@ fn run_shell(vm: &VirtualMachine, scope: Scope) -> PyResult<()> {
                         ..
                     }) => {
                         continuing = true;
-                        continue;
                     }
                     _ => {
                         input = String::new();
                     }
                 }
             }
             Err(ReadlineError::Interrupted) => {
-                // TODO: Raise a real KeyboardInterrupt exception
-                println!("^C");
+                let exc = vm
+                    .new_empty_exception(vm.ctx.exceptions.keyboard_interrupt.clone())
+                    .unwrap();
+                print_exception(vm, &exc);
```

Comment:
There is something which occurs strange to me when examining this piece of code. The exception is created and printed, but should it not be inserted into the virtualmachine in some way? Maybe we can inject it into the virtual machine in a similar manner in which the signals are inserted into the virtualmachine, with the check_signals function and some sort of a queue.

I would suggest to add a function to the virtualmachine, named something like `inject_exception`, which will put this exception object into a queue, or a slot, and the frame eval loop will pick it up.

Reply:
The problem here is that the code I was converting from CPython here is actually from the builtin `input()` function, which then gets used by the `code` Python module to run the shell. Would it be okay if I left it like this for now but added `input()` and `code.py` in a separate PR?

Reply:
Actually no, it doesn't use `code.py`, that's just trying to emulate it. I'll look into this further.